web/libs/editor/src/components/InteractiveOverlays/CommentsOverlay.tsx

```diff
@@ -0,0 +1,193 @@
+import { observer } from "mobx-react";
+import { isAlive } from "mobx-state-tree";
+import type React from "react";
+import { memo, useCallback, useEffect, useMemo, useRef, useState } from "react";
+import { useMounted } from "../../common/Utils/useMounted";
+
+import ResizeObserver from "../../utils/resize-observer";
+import { guidGenerator } from "../../utils/unique";
+
+import styles from "./CommentsOverlay.module.scss";
+
+import NodesConnector from "./NodesConnector";
+
+const CommentIcon = () => {
+  return (
+    <g className={styles.commentIcon}>
+      <path
+        className={styles.commentIconBackground}
+        d="M0 12C0 5.3726 5.3726 0 12 0C18.6274 0 24 5.3726 24 12C24 18.6274 18.6274 24 12 24H0V12Z"
+      />
+      <path
+        className={styles.commentIconLines}
+        d="M18 8V9.3333H6V8H18ZM6 16H12V14.6667H6V16ZM6 12.6667H18V11.3333H6V12.6667Z"
+      />
+    </g>
+  );
+};
+
+const ICON_SIZE = 24;
+const PADDING_COMPENSATION = 3;
+const COMMENT_ICON_OVERLAP = 4;
+type CommentItemProps = {
+  comment: MSTComment;
+  rootRef: React.MutableRefObject<HTMLOrSVGElement | undefined>;
+};
+const CommentItem: React.FC<CommentItemProps> = observer(({ comment, rootRef }) => {
+  const root = rootRef.current;
+  const node = comment.regionRef?.overlayNode;
+  const isHidden = !node || node.hidden;
+  const [forceUpdateId, forceUpdate] = useState<any>({});
```

Comment:
that's not obvious that empty objects you use here are used because they are different every render and setting another empty object will actually re-trigger rendering. maybe a comment at least?

Reply:
That's not my solution to be honest 😅
https://github.com/HumanSignal/label-studio/blob/develop/web/libs/editor/src/components/RelationsOverlay/RelationsOverlay.jsx#L109